evaluations-service-core/evaluations-service-models/src/main/java/ar/edu/itba/cep/evaluations_service/models/ExerciseSolutionResult.java

```diff
@@ -0,0 +1,117 @@
+
+/**
+ * Represents an exercise's solution result (i.e approved or failed).
+ * This class relates an {@link ExerciseSolution} and a {@link TestCase},
+ * indicating whether the solution is approved or failed, when testing it with the test case.
+ */
+public class ExerciseSolutionResult {
+
+    /**
+     * The exercise's solution result id.
+     */
+    private final long id;
+    /**
+     * The {@link ExerciseSolution} to which this result makes reference.
+     */
+    private final ExerciseSolution solution;
+    /**
+     * The test case being used to reach the result.
+     */
+    private final TestCase testCase;
+    /**
+     * Indicates whether the result is approved or failed.
+     */
+    private final Result result;
+
+
+    /**
+     * Constructor.
+     *
+     * @param solution The {@link ExerciseSolution} to which this result makes reference.
+     * @param testCase The test case being used to reach the result.
+     * @param result   Indicates whether the result is approved or failed.
+     */
+    public ExerciseSolutionResult(final ExerciseSolution solution, final TestCase testCase, final Result result) {
+        this.result = result;
+        this.id = 0;
+        this.solution = solution;
+        this.testCase = testCase;
+    }
+
+
+    /**
+     * @return The exercise's solution result id.
+     */
+    public long getId() {
+        return id;
+    }
+
+    /**
+     * @return The {@link ExerciseSolution} to which this result makes reference.
+     */
+    public ExerciseSolution getSolution() {
+        return solution;
+    }
+
+    /**
+     * @return The test case being used to reach the result.
+     */
+    public TestCase getTestCase() {
+        return testCase;
+    }
+
+    /**
+     * @return Indicates whether the result is approved or failed.
+     */
+    public Result getResult() {
+        return result;
+    }
+
+    // ================================
+    // equals, hashcode and toString
+    // ================================
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof ExerciseSolutionResult)) {
+            return false;
+        }
+        final var that = (ExerciseSolutionResult) o;
+        return id == that.id;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id);
+    }
+
+    @Override
+    public String toString() {
+        return "ExerciseSolutionResult [" +
+                "ID: " + id + ", " +
+                "Solution: " + solution + ", " +
+                "TestCase: " + testCase + ", " +
+                "Result: " + result +
+                "]";
+    }
+
+
+    // ================================
+    // Helpers
+    // ================================
+
+    /**
+     * An enum indicating whether the result is "approved" or "failed".
+     */
+    public enum Result {
+        APPROVED,
+        FAILED,
+        ;
```

Comment:
does this compile? `,;`?

Reply:
Yes, enums and arrays can have a ```,``` in the end, and the ```;``` can also be present.

Reply:
I'd try to make the `description`, `duration` and `startingAt` compulsory for the creation of the object, somehow.

Reply:
Yes, in another PR